CustomTabBar/TabButton.swift

```diff
@@ -165,6 +187,24 @@ private extension TabButton {
         view.trailingAnchor.constraint(equalTo: trailingAnchor, constant: margin ? -TabButton.margin : 0).isActive = true
         view.bottomAnchor.constraint(equalTo: bottomAnchor, constant: margin ? -TabButton.margin : 0).isActive = true
     }
+
+    func addBadgeLabel() {
+        guard badgeLabel.superview == nil else { return }
+        addSubview(badgeLabel)
+        badgeLabel.translatesAutoresizingMaskIntoConstraints = false
+        if let imageAnchor = imageButton.imageView?.trailingAnchor {
+            badgeLabel.centerXAnchor.constraint(equalTo: imageAnchor, constant: TabButton.badgeMargin).isActive = true
+        } else {
+            badgeLabel.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -TabButton.badgeMargin).isActive = true
+        }
+        badgeLabel.topAnchor.constraint(equalTo: topAnchor, constant: TabButton.badgeMargin).isActive = true
+        let badgeHeight = badgeLabel.intrinsicContentSize.height * 1.1
```

Comment:
Do we need to do a layout pass before this has useful information? Or does this already get the right thing by setting the content?